src/VarCache.ts

```diff
@@ -33,11 +33,13 @@ export default class VarCache {
   public token: string = ''
   public variants = []
 
+  public constructor(private internalState: InternalState) {}
+
   public applyDiffs(diffs, variants, actionMetadata) {
     this.diffs = diffs
     this.variants = variants
     this.actionMetadata = actionMetadata
-    InternalState.hasReceivedDiffs = true
+    this.internalState.hasReceivedDiffs = true
```

Comment:
This field is only set here, and is checked only in the VarCache events (variablesChangedHandlers). I believe it's a better idea to move the event to this class and keep the field `private`.

Reply:
Nice catch, updated as suggested!